CommonTools/PileupAlgos/plugins/PuppiPhoton.cc

```diff
@@ -76,7 +71,25 @@ void PuppiPhoton::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
   edm::Handle<CandidateView> hPuppiProduct;
   iEvent.getByToken(tokenPuppiCandidates_,hPuppiProduct);
   const CandidateView *pupCol = hPuppiProduct.product();
-  for(CandidateView::const_iterator itPho = phoCol->begin(); itPho!=phoCol->end(); itPho++) {
+  if(usePFphotons_) {
+   for(CandidateView::const_iterator itPho = pfCol->begin(); itPho!=pfCol->end(); itPho++) {
```

Comment:
hi @ahinzmann - this can instead be a range based loop

for ( const auto & pho : pfCol ) {...
 if pho.pt() < pt_ continue...